mllib/src/test/scala/org/apache/spark/ml/classification/LinearSVCSuite.scala

```diff
@@ -127,6 +127,27 @@ class LinearSVCSuite extends SparkFunSuite with MLlibTestSparkContext with Defau
     MLTestingUtils.checkCopyAndUids(lsvc, model)
   }
 
+  test("LinearSVC threshold acts on rawPrediction") {
+    val lsvc =
+      new LinearSVCModel(uid = "myLSVCM", coefficients = Vectors.dense(1.0), intercept = 0.0)
+    val df = spark.createDataFrame(Seq(
+      (1, Vectors.dense(1e-7)),
+      (0, Vectors.dense(0.0)),
+      (-1, Vectors.dense(-1e-7)))).toDF("id", "features")
+
+    def checkResults(threshold: Double, expected: Set[(Int, Double)]): Unit = {
+      lsvc.setThreshold(threshold)
+      val results = lsvc.transform(df).select("id", "prediction").collect()
+        .map(r => (r.getInt(0), r.getDouble(1)))
+        .toSet
+      assert(results === expected, s"Failed for threshold = $threshold")
+    }
+
+    checkResults(0.0, Set((1, 1.0), (0, 0.0), (-1, 0.0)))
```

Comment:
It might be a good idea to additionally test the other code path, by calling `lsvc.set(lsvc.rawPrediction, "")`. The way it is, we don't know if `predict` works correctly (this is done in the logistic regression suite).